patient-care/webapp/client/manageObjects/manageObjects.js

```diff
@@ -137,48 +137,55 @@ Template.manageObjectsGrid.onRendered(function () {
   // Make the manage object detail sticky so it moves down
   // the page as the user scrolls down.
   function refreshSticky() {
-    // Only activate the sticky when the height of the master list is
-    // greater than a pageful. Don't have a sticky when creating a new object
-    // because it's too annoying to keep refreshing the sticky.
-    // (Creation UIs tend to get bigger as the user fills in information.)
-    if (instance.$("#manage-obj-master").height() > $(window).height() &&
+    // Only activate the sticky when the height of the master list is larger
+    // than that of the detail view. Semantic UI doesn't handle the sticky
+    // well when the stickied content is taller than the wrapper.
+    // Once a sticky has been destroyed for a route (it got taller than
+    // the master list), don't watch to see if it gets shorter again. Instead,
+    // wait until the selected item changes to check if we should recreate
+    // the sticky.
+    // Don't have a sticky when creating a new object because those UIs tend
+    // to grow as the user fills in information.
+
+    let masterHeight = instance.$("#manage-obj-master").height();
+    let detailHeight = instance.$("#manage-obj-detail").height();
+
+    if (detailHeight < masterHeight &&
         FlowRouter.getParam("selected") !== "create") {
-      if (stickied) {
-        // refresh the sticky because it's been set up already
-        instance.$("#manage-obj-detail").sticky("refresh");
-      } else {
+      if (!stickied) {
         // set up the sticky for the first time
         instance.$("#manage-obj-detail").sticky({
           context: "#manage-obj-context",
           offset: 55,
+          observeChanges: true,
+          onReposition() {
+            // NOTE: recursive
+            Meteor.defer(refreshSticky);
+          },
         });
-      }
 
-      stickied = true;
+        stickied = true;
+      }
     } else {
       // destroy the sticky
       instance.$("#manage-obj-detail").sticky("destroy");
 
+      // set some CSS that isn't cleared when the sticky is destroyed
+      instance.$("#manage-obj-detail-container").css("min-height", "auto");
+
       stickied = false;
     }
   }
 
-  // call refreshSticky when a reactive variable changes
+  // watch these reactive functions so it'll update
+  // when they change, but don't do anything with the return value
   instance.autorun(() => {
-    // reactively watch the count of the items on the left so it'll update
-    // when it changes, but don't do anything with the return value
     getObjects(instance).count();
 
-    Meteor.defer(refreshSticky);
-  });
+    FlowRouter.getParam("selected");
 
-  // only call it 200 milliseconds after the thing has stopped resizing
-  instance.debouncedRefreshSticky = _.debounce(function() {
     Meteor.defer(refreshSticky);
-  }, 200);
-
-  // call refreshSticky when the window is resized
-  $(window).resize(instance.debouncedRefreshSticky);
```

Comment:
We no longer use the window height to determine when to show the sticky, so we don't need this anymore.